kview/src/main/java/dev/ikm/komet/kview/events/EditConceptFullyQualifiedNameEvent.java

```diff
@@ -17,38 +17,60 @@
 
 import dev.ikm.komet.framework.events.Evt;
 import dev.ikm.komet.framework.events.EvtType;
+import dev.ikm.komet.framework.view.ViewProperties;
+import dev.ikm.komet.kview.mvvm.model.DescrName;
 import dev.ikm.tinkar.common.id.PublicId;
 
 public class EditConceptFullyQualifiedNameEvent extends Evt  {
 
     public static final EvtType<EditConceptFullyQualifiedNameEvent> EDIT_FQN = new EvtType<>(Evt.ANY, "EDIT_FQN");
 
-    private PublicId publicId;
+    private final PublicId publicId;
 
+    private final DescrName descrName;
 
-    public EditConceptFullyQualifiedNameEvent(Object source, EvtType eventType) {
+    private final ViewProperties viewProperties;
+    /**
+     * Constructs a prototypical Event.
+     *
+     * @param source    the object on which the Event initially occurred
+     * @param eventType the edit event type
+     * @param viewProperties the view properties for UI.
+     * @param publicId publicId of the concept being edited.
+     * @throws IllegalArgumentException if source is null
+     */
+    public EditConceptFullyQualifiedNameEvent(Object source, EvtType eventType, ViewProperties viewProperties, PublicId publicId) {
         super(source, eventType);
+        this.viewProperties = viewProperties;
+        this.publicId = publicId;
+        this.descrName = null;
     }
+
     /**
      * Constructs a prototypical Event.
      *
      * @param source    the object on which the Event initially occurred
-     * @param eventType
-     * @param publicId
+     * @param eventType the edit event type
+     * @param viewProperties viewProperties for the display
+     * @param descrName the model object.
      * @throws IllegalArgumentException if source is null
      */
-    public EditConceptFullyQualifiedNameEvent(Object source, EvtType eventType, PublicId publicId) {
+    public EditConceptFullyQualifiedNameEvent(Object source, EvtType eventType, ViewProperties viewProperties, DescrName descrName) {
```

Comment:
Can we not pass view properties?
All JavaFX controllers have a view properties. The objective of eventing is try to keep the payload (things to transmit) small.

Reply:
The viewProperties are not set while editing the concept without transactions, the value is null. Had to pass it in one of the methods.